### PR TITLE
folly 0.47.0

### PR DESCRIPTION
**Pull request by marpaia**
I took this over from #37982 and addressed the comments in the PR.

### CODE REVIEW

File: Library/Formula/folly.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        folly::fbvector<int> numbers({0, 1, 2, 3});
+        numbers.reserve(10);
+        for (int i = 4; i < 10; i++) {
+          numbers.push_back(i * 2);
+        }
+        assert(numbers[6] == 12);
+        return 0;
+      }
+    EOS
+    system ENV.cxx, "-std=c++11", "test.cc", "-L#{lib}", "-lfolly", "-o", "test"
```
Comment (mistydemeo) [inline Library/Formula/folly.rb]: Does folly itself need C++11 to build?

**Comment by mistydemeo** [comment]
Thanks for picking this up!

File: Library/Formula/folly.rb (hunk 1 above)
Comment (marpaia) [inline Library/Formula/folly.rb]: yup

**Comment by mistydemeo** [comment]
Instead of a 10.9 requirement, add `needs :cxx11` below the dependencies and `ENV.cxx11` at the top of the install method; that'll allow 10.8's clang to build it. It should also work on older OS X versions with a C++11-capable compiler.

**Comment by marpaia** [comment]
@mistydemeo updated the commit. just `ENV.cxx11`? not `ENV.cxx11 = true` or anything?

**Comment by mistydemeo** [comment]
Yep, that's right. It's a method call that sets environment variables.

**Comment by marpaia** [comment]
ah, that's right, i forgot that you can omit the `()` :weary:

File: Library/Formula/folly.rb [hunk 2]
```diff
@@ -0,0 +1,45 @@
+class Folly < Formula
+  desc "An open-source C++ library developed and used at Facebook."
+  homepage "https://github.com/facebook/folly"
+  url "https://github.com/facebook/folly.git", :revision => "6457c4429379011d12ee841e8325dddeebec538e"
```
Comment (guoxiao) [inline Library/Formula/folly.rb]: Should the url be `https://github.com/facebook/folly/archive/v0.45.0.tar.gz` ?

Comment (marpaia) [inline Library/Formula/folly.rb]: the build doesn't work on OS X, it has since been fixed in master, but a release hasn't been cut

Comment (mistydemeo) [inline Library/Formula/folly.rb]: Can you use the tarball and backport the patches that fix it?

Comment (marpaia) [inline Library/Formula/folly.rb]: i'm going to see if we can just cut a new release which builds on OS X.

**Comment by marpaia** [comment]
@mistydemeo the build still fails on ML. I sincerely doubt that Folly maintainers have any desire whatsoever to support ML, especially given that it doesn't even receive security updates anymore. is it cool if i just require mavericks?

**Comment by mistydemeo** [comment]
Yep, that's fine.

**Comment by marpaia** [comment]
so we cut a new folly release at the commit hash that has previously built fine on your 10.9 and 10.10 jenkins, removed 10.8 support, building from the tar, etc. should be good to go, but i'm not sure why jenkins isn't picking up the build. perhaps i've `git commit --amend` too much?

File: Library/Formula/folly.rb [hunk 3]
```diff
@@ -0,0 +1,47 @@
+class Folly < Formula
+  desc "An open-source C++ library developed and used at Facebook."
+  homepage "https://github.com/facebook/folly"
+  url "https://github.com/facebook/folly/archive/v0.47.0.tar.gz"
+  sha256 "7d3309088cae1817d60996611e5fc0825c5c8df36d244f247c1d49ebc38c35fb"
+  version "0.47.0"
```
Comment (mistydemeo) [inline Library/Formula/folly.rb]: Can remove the explicit version now that you're using a tagged release; it can be inferred from the tarball name.

**Comment by DomT4** [comment]
@BrewTestBot test this please

File: Library/Formula/folly.rb [hunk 4]
```diff
@@ -0,0 +1,46 @@
+class Folly < Formula
+  desc "An open-source C++ library developed and used at Facebook."
```
Comment (mikemcquaid) [inline Library/Formula/folly.rb]: What does it do, though?

**Comment by mikemcquaid** [comment]
Happy to merge this; just need a slightly better description.

File: Library/Formula/folly.rb (hunk 4 above)
Comment (marpaia) [inline Library/Formula/folly.rb]: That's actually the "official" description of the project. I think the ambiguity is intentional, its supposed to feel like a grab bag of awesome C++ utilities which we use at FB.

Comment (mikemcquaid) [inline Library/Formula/folly.rb]: Could you make it a little less ambiguous? Maybe even just `grab bag of awesome` etc.

**Comment by marpaia** [comment]
@mikemcquaid i changed it based on the second sentence of the release blog post: https://www.facebook.com/notes/facebook-engineering/folly-the-facebook-open-source-library/10150864656793920

i didn't want to get too detailed, as i didn't want the desc line to get too long, but if you'd prefer something else for the description, i'm down for whatever works for you all.

**Comment by mikemcquaid** [comment]
@marpaia It's great, thanks!

**Comment by marpaia** [comment]
omg desc is too long now. updating.

**Comment by DomT4** [comment]
Merged in https://github.com/Homebrew/homebrew/commit/3ef948bb3484c2af701c8a226550d152d8636f64. Thank you for your contribution to Homebrew, we appreciate it! :smiley_cat:.

It's a nit but just for future reference new formulae commit messages should be styled `name 1.2.3 (new formula)`, Thanks!

**Comment by marpaia** [comment]
cheers, thanks!

**Pull request closed without merging**